src/condition.cpp

```diff
@@ -1427,6 +1427,29 @@ conditional_t::func f_npc_train_spells( bool is_npc )
     };
 }
 
+conditional_t::func f_follower_present(const JsonObject &jo, std::string_view member)
+{
+    dbl_or_var empty;
+    const std::string &var_name = jo.get_string(std::string(member));
+    return [var_name](dialogue const &d)
+    {
+        shared_ptr_fast<npc> npc_to_get = overmap_buffer.find_npc_by_unique_id(std::string(var_name));
```

Comment:
Unique_ID is only used to track the refugee center guards, you'd want to compare their class IDs like this

```suggestion
npc *npc_to_get;
for( npc &guy : all_npcs() ) {
     if ( guy.myclass == var_name ) {
     npc_to_get = guy;
     break;
     }
```

Reply:
thanks, done